cfpb/cfgov-refresh
# Add new EnforcementActionPage with metadata

Currently, we use the Document Detail page type to create pages about Enforcement Actions in Wagtail, and use the Related Metadata streamfield block to store important information about the action. This means that the metadata (which should be present on all enforcement actions) might be missing or malformed, and is not easily searchable or filterable.

This PR adds an Enforcement Action page type, which includes a new tab in the wagtail interface called "Metadata". It contains database fields for the required metadata. This will allow us to filter on these values in the Enforcement Actions filtered list page in a future update.


## Additions

- `EnforcementActionPage` page type
- `EnforcementActionStatus` class to allow an Enforcement Action to have an arbitrary number of statuses
- Templates and styling for the new page type

## Testing

1. Running this branch, log in to the [Wagtail admin](http://localhost:8000/admin).
2. Create a new page of the Enforcement Action Page type.
    1. Be sure to fill in the information in the metadata panel
    2. Publish the page
3. Visit the page you created, see that the metadata are displayed correctly in the sidebar of the published page.

## Screenshots
The Metadata panel on the published Enforcement Action page:
<img width="215" alt="Screen Shot 2020-02-27 at 12 31 52 PM" src="https://user-images.githubusercontent.com/4295388/75469534-2ffa8800-595d-11ea-8dbd-53fef696a8fc.png">

The Metadata panel on the editor's view in Wagtail:
<img width="500" alt="Screen Shot 2020-02-27 at 2 50 17 PM" src="https://user-images.githubusercontent.com/4295388/75484833-84aafc80-5977-11ea-8afb-d05e86cfef67.png">


## Todos

- ~~Styling for enforcement action status in sidebar (@wpears)~~
- ~~Make the migration Wagtail 2 compatible (Betsy)~~
- ~~Require at least one Category and Enforcement Status~~
- Require at least one Topic
- ~~Require `date_filed` field~~

In a future update, we will migrate existing Enforcement Actions (currently stored in `DocumentDetailPage`) to the new `EnforcementActionPage`

## Checklist


## Testing checklist

### Browsers


### Accessibility


### Other
